### PR TITLE
chore(API): increase uniformity of dunders

### PR DESCRIPTION
**Pull request by d-bohls**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nixnet-python/blob/master/CONTRIBUTING.rst).
- [x] New tests have been created for any new features or regression tests for bugfixes.
- [x] `tox` successfully runs, including unit tests and style checks (see [CONTRIBUTING.md](https://github.com/ni/nixnet-python/blob/master/CONTRIBUTING.rst)).

### CODE REVIEW

File: nixnet/database/_signal.py [hunk 1]
```diff
@@ -13,8 +13,9 @@ class Signal(object):
     """Database signal"""
 
     def __init__(self, handle):
+        # type: (int) -> None
         self._handle = handle
-        self._dbc_attributes = None
+        self._dbc_attributes = None  # type: typing.Union[None, _dbc_attributes.DbcAttributeCollection]
```
Comment (epage) [inline nixnet/database/_signal.py]: You can use `Optional` instead

https://docs.python.org/3/library/typing.html#typing.Optional

File: nixnet/database/_cluster.py [hunk 2]
```diff
@@ -21,8 +21,9 @@ class Cluster(object):
     """Database cluster"""
 
     def __init__(self, handle):
+        # type: (int) -> None
         self._handle = handle
-        self._dbc_attributes = None
+        self._dbc_attributes = None  # type: typing.Union[None, _dbc_attributes.DbcAttributeCollection]
```
Comment (d-bohls) [inline nixnet/database/_cluster.py]: Adding a type annotation to this constructor causes mypy to start tracking other variables in init. mypy then began complaining about the type transition of `self._dbc_attributes` from `None` to `_dbc_attributes.DbcAttributeCollection`.

> Incompatible types in assignment (expression has type 'DbcAttributeCollection' variable has type None)

That's why I had to add a type annotation to this line (also done for other classes with DBC attributes) .

File: nixnet/database/_signal.py (hunk 1 above)
Comment (d-bohls) [reply nixnet/database/_signal.py]: I like that more. I'll do that.

File: nixnet/database/_cluster.py (hunk 2 above)
Comment (epage) [reply nixnet/database/_cluster.py]: btw to make the mypy typing more correct for container functions, I think you can switch from inheriting from `abc.Mapping` (or whatever) to `typing.Mapping[From, To]`.  Not sure if this works for Python2.

Example:
```python
T = TypeVar('T')

class MyDict(Mapping[str, T]):
    ...
```
https://docs.python.org/3/library/typing.html#user-defined-generic-types

**Pull request merged**